plugin.rb

```diff
@@ -71,6 +71,14 @@ def response(env)
   end
 end
 
+# You should use this register if you want to add custom paths to traverse the user details JSON. 
+# The path's last attribute will be the attribute name, which we'll store in the user 
+# associated account extra attribute.
+#
+# For example, the `account.custom_attribute` value is stored in the associated account 
+# extra field with custom_attribute as the name.
+DiscoursePluginRegistry.define_filtered_register :custom_user_details
```

Comment:
I think we should name this something plugin-specific, to make sure we don't get a clash in future. Maybe `:oauth2_basic_additional_json_paths`?